examples/lib/stories/components/composability.dart

```diff
@@ -27,7 +27,7 @@ class ParentSquare extends Square {
       Square(Vector2(70, 200), Vector2(50, 50), angle: 5),
     ];
 
-    children.forEach(addChild);
+    children.forEach((c) => addChild(c, gameRef: gameRef));
```

Comment:
this was actually a bug. if any child had a HasGameRef it would not be populated. my new assert catches that.

Reply:
no, but they could

Reply:
But this is an example that the writer of the example is in total control over writing that game? I don't see the need to give the children a `gameRef` when they don't need it?

Reply:
that is true, but I thought it would lead to confusing behavior to allow this, might cause not so clever users to get into weird states and think it's a bug

Reply:
If that is the case, maybe we should consider having the `gameRef` on all `BaseComponent` as default instead?

Reply:
Ah right, that wouldn't solve the case when children are added to the component before the component is added to the game though...

Reply:
I think we can improve that on followups, it's not directly related to this PR

Reply:
we only need to pass when the parent doesn't have it